src/bylines/index.js

```diff
@@ -3,64 +3,520 @@
 /**
  * WordPress dependencies
  */
-import { ToggleControl, TextareaControl } from '@wordpress/components';
+import { Button, Modal, ToggleControl } from '@wordpress/components';
 import { useSelect, useDispatch } from '@wordpress/data';
 import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
 import { __ } from '@wordpress/i18n';
 import { registerPlugin } from '@wordpress/plugins';
+import apiFetch from '@wordpress/api-fetch';
+import { Icon, plus } from '@wordpress/icons';
+import { store as coreStore } from '@wordpress/core-data';
+
 /**
  * External dependencies
  */
-import { useEffect, useState } from 'react';
+import { useEffect, useState, useRef } from 'react';
+
 /**
  * Internal dependencies
  */
 import './style.scss';
 
-const BYLINE_ID = 'newspack-byline';
+const BASE_QUERY = {
+	_fields: 'id,name',
+	context: 'view', // Allows non-admins to perform requests.
+};
+
+const AUTHORS_QUERY = {
+	who: 'authors',
+	per_page: 100,
+	...BASE_QUERY,
+};
+
+/** Close icon copied from @wordpress/icons/src/library/close.js to be used as markup */
+const close = `
+	<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+		<path d="M13 11.8l6.1-6.3-1-1-6.1 6.2-6.1-6.2-1 1 6.1 6.3-6.5 6.7 1 1 6.5-6.6 6.5 6.6 1-1z" />
+	</svg>
+`;
+
+const CustomBylineModal = ( { children } ) => {
+	const [ isOpen, setOpen ] = useState( false );
+	const openModal = () => setOpen( true );
+	const closeModal = () => setOpen( false );
+
+	return (
+		<>
+			<Button variant="secondary" onClick={ openModal }>
+				Set Custom Byline
+			</Button>
+			{ isOpen && (
+				<Modal title="Set Custom Byline" onRequestClose={ closeModal }>
+					{ children }
+					<Button variant="secondary" onClick={ closeModal }>
+						Close
+					</Button>
+				</Modal>
+			) }
+		</>
+	);
+};
+const BylineTextarea = ( { byline, onRendered } ) => {
+	const { getEditedPostAttribute } = useSelect( select =>
+		select( 'core/editor' )
+	);
+
+	/** The custom byline stored as meta */
+	const metaByline =
+		getEditedPostAttribute( 'meta' )[ newspackBylines.metaKeyByline ] || '';
+
+	/**
+	 * Parse byline meta to convert custom tags (<Author></Author> or [Author][/Author]) to token markup.
+	 *
+	 * @see    {@link https://github.com/Automattic/newspack-plugin/tree/trunk/includes/bylines#readme|Custom Bylines}
+	 * @return {string} Parsed byline looking up for <Author id=1></Author> tags and replacing them.
+	 */
+	const bylineParser = () => {
+		const tokenMarkup = `<span id="token-$1" class="components-form-token-field__token token-inline-block author-token" data-token="$1">
```

Comment:
This looks like the same component as `TokenInlineBlock` - can they be refactored to use the same function for generating markup?

Reply:
Yes, it is, but I couldn't think about how to retrieve the render of a component to store on a variable, I guess that's not possible, but another solution could be possible, I'm trying to think what.